services/blockchain-indexer/shared/indexer/transactionProcessor/nft/transfer.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const {
+	Logger,
+	DB: { MySQL: { getTableInstance } },
+} = require('lisk-service-framework');
+const config = require('../../../../config');
+
+const { TRANSACTION_STATUS } = require('../../../constants');
+
+const logger = Logger();
+
+const MYSQL_ENDPOINT = config.endpoints.mysql;
+
+const { indexAccountAddress } = require('../../accountIndex');
+
+const nftTableSchema = require('../../../database/schema/nft');
+const transactionsTableSchema = require('../../../database/schema/transactions');
+
+const getNFTTable = () => getTableInstance(nftTableSchema, MYSQL_ENDPOINT);
+const getTransactionsTable = () => getTableInstance(transactionsTableSchema, MYSQL_ENDPOINT);
+
+// Command specific constants
+const COMMAND_NAME = 'transfer';
+
+// eslint-disable-next-line no-unused-vars
+const applyTransaction = async (blockHeader, tx, events, dbTrx) => {
+	logger.trace(`Updating index for the account with address ${tx.params.recipientAddress} asynchronously.`);
+	indexAccountAddress(tx.params.recipientAddress);
+
+	if (tx.executionStatus !== TRANSACTION_STATUS.SUCCESSFUL) return;
+
+	const nftTable = await getNFTTable();
+	const transactionsTable = await getTransactionsTable();
+
+	tx = {
+		...tx,
+		...tx.params,
+	};
+
+	const searchParams = {
+		nftID: tx.nftID,
+	};
+	const [nft] = await nftTable.find(searchParams, ['nftID', 'owner', 'escrowChainID']);
+
+	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
+	nft.owner = tx.recipientAddress;
+	await nftTable.upsert(nft, dbTrx);
```

Comment:
I think there is no need to call `db.find` here. Instead we can directly `upsert`
```suggestion
	logger.trace(`Updating owner of nft ${tx.nftID} to ${tx.recipientAddress}.`);
	const nft = {
	   nftID: tx.nftID,
	   owner: tx.recipientAddress,
	};
	await nftTable.upsert(nft, dbTrx);
```